smartsim/_core/launcher/colocated.py

```diff
@@ -142,17 +145,86 @@ def _build_colocated_wrapper_cmd(port=6780,
     ])
     for db_arg, value in extra_db_args.items():
         # replace "_" with "-" in the db_arg because we use kwargs
-        # for the extra configurations and Python doesn't allow a hypon
+        # for the extra configurations and Python doesn't allow a hyphen
         # in a variable name. All redis and KeyDB configuration options
         # use hyphens in their names.
         db_arg = db_arg.replace("_", "-")
         db_cmd.extend([
             f"--{db_arg}",
             value
         ])
+
+    db_models = kwargs.get("db_models", None)
+    if db_models:
+        db_model_cmd = _build_db_model_cmd(db_models)
+        db_cmd.extend(db_model_cmd)
+    
+    db_scripts = kwargs.get("db_scripts", None)
+    if db_scripts:
+        db_script_cmd = _build_db_script_cmd(db_scripts)
+        db_cmd.extend(db_script_cmd)
+
     # run colocated db in the background
     db_cmd.append("&")
 
     cmd.extend(db_cmd)
     return " ".join(cmd)
 
+
+def _build_db_model_cmd(db_models):
+    cmd = []
+    for db_model in db_models:
+        cmd.append("+db_model")
+        cmd.append(f"--name={db_model.name}")
+        if db_model.file:
+            cmd.append(f"--file={db_model.file}")
+        else:
+            err_msg = "ML model can not be set from memory for colocated databases.\n"
```

Comment:
can we detect this prior to launch? The ``Model`` object should be able to detect this when the user adds it to a colocated model and we can raise a ``ValueError``

Reply:
Yes, this was easy with models, but it became a bit more difficult with ensembles. Done it for all cases.